keadm/app/cmd/util/edgecoreinstaller.go

```diff
@@ -64,51 +74,137 @@ func (ku *KubeEdgeInstTool) createEdgeConfigFiles() error {
 		return fmt.Errorf("not able to create %s folder path", KubeEdgeConfPath)
 	}
 
-	//Create edge.yaml
-	//Update edge.yaml, node ip to act as node id
-	//Get node ip address from the interface
-	edgeIP, err := GetInterfaceIP()
-	if err != nil {
-		return err
+	// //Create edge.yaml
+	//Update edge.yaml with a unique id against node id
+	//If the user doesn't provide any edge ID on the command line, then it generates unique id and assigns it.
+	edgeID := uuid.New().String()
+	if "" != ku.EdgeNodeID {
```

Comment:
s/"" != ku.EdgeNodeID/ku.EdgeNodeID != ""
Applicable to similar places

Reply:
Why?

Reply:
It is not explicitly mentioned in any coding guidelines but usually variable is checked against the constant. For example ``` err != nil ``` I doubt we will find ``` nil != err ```

Reply:
Yes, as it is not mentioned in coding guidelines. This is fine. BTB i kind of followed C coding best practices, where it says that in a check, lhs is preferred to be a constant and rhs the compared variable. Hence applied the same and it is obvious it doesn't break any logic. So lets keep it as it is.